src/pkg/packager/deploy.go

```diff
@@ -346,7 +346,9 @@ func (d *deployer) deployComponent(ctx context.Context, pkgLayout *layout.Packag
 
 	if hasImages {
 		imagePushOpts := ImagePushOptions{
-			Cluster:         d.c,
+			Cluster: d.c,
+			// we only want to push the images for this single component
+			Components:      []v1alpha1.ZarfComponent{component},
```

Comment:
Thoughts on having deploy call `images.Push` directly? That's how it originally was before I changed it, introducing the bug. I'm thinking through the user flows, and I think almost always when someone wants to push only images of certain components it'd be during deploy, or they'd be using a `filter` to get the components they want.